### PR TITLE
Rework Maven Source Plugin Skip

### LINKED ISSUE #11245: [BUG] mvn verify for the source-javadoc fails and no pre-merge check catches it
**Describe the bug**

Switching to the forked version of sorces jar creation in #11237 made it necessary to skip its execution manually in  the source-less module such as `aggregator` and `dist`. However, it ended up  in a bug that disabled dist jar creation. 

A build assert in the build and integration tests should have caught it, however, the `source-javadoc` profile is not exercised until the nightly CI build

**Steps/Code to reproduce bug**
```
$ mvn -Psource-javadoc clean verify -pl dist -am -DskipTests -Dbuildver=331
... rapids-4-spark_2.12-24.08.0-SNAPSHOT-cuda11.jar!/META-INF/maven/com.nvidia/rapids-4-spark_2.12/pom.xml doesn't exist
``` 

**Expected behavior**
Should be impossible to break the build like #11237

### PR DESCRIPTION
**Pull request by gerashegalov**
Fixes #11245 

- Follow the existing pattern of `rapids.*.jar.phase` 
- Include source-javadoc to the verify PR check

### CODE REVIEW

**Review by jlowe — DISMISSED** [review]

**Comment by gerashegalov** [comment]
build

File: aggregator/pom.xml [hunk 1]
```diff
@@ -44,6 +44,7 @@
         <rapids.default.jar.phase>initialize</rapids.default.jar.phase>
         <!-- Maven to register attached artifact , which we later replace -->
         <rapids.shim.jar.phase>initialize</rapids.shim.jar.phase>
+        <rapdis.source.jar.phase>none</rapdis.source.jar.phase>
```
Comment (razajafri) [inline aggregator/pom.xml:47]: typo

File: dist/pom.xml [hunk 2]
```diff
@@ -46,6 +46,7 @@
         <dist.jar.pom.url>jar:file:${dist.jar.name}!/META-INF/maven/${project.groupId}/${project.artifactId}/pom.xml</dist.jar.pom.url>
         <rapids.default.jar.phase>none</rapids.default.jar.phase>
         <rapids.jni.unpack.skip>false</rapids.jni.unpack.skip>
+        <rapdis.source.jar.phase>none</rapdis.source.jar.phase>
```
Comment (razajafri) [inline dist/pom.xml:49]: same comment as above

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Review by pxLi — APPROVED** [review]

**Pull request merged**